.github/workflows/get_charms_build_with_cache.yaml

```diff
@@ -0,0 +1,32 @@
+name: Get charms and build with cache (common)
+
+on:
+  workflow_call:
+
+jobs:
+
+  get-charm-paths:
+    name: Generate the Charm Matrix content
+    runs-on: ubuntu-latest
+    outputs:
+      charm_paths: ${{ steps.get-charm-paths.outputs.charm-paths }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Get paths for all charms in this repo
+        id: get-charm-paths
+        uses: canonical/kubeflow-ci/actions/get-charm-paths@main
+
+  build:
+    name: Build charms
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v21.0.0
+    needs: get-charm-paths
+    strategy:
+      fail-fast: false
```

Comment:
question: can you explain why this is needed?

Reply:
In some cases with parallel steps, if one fails, the rest will be cancelled. We do not want all of them to be cancelled just right away.

Reply:
good point, thanks for the clarification!